cmd/influxd/run.go

```diff
@@ -43,54 +43,32 @@ func execRun(args []string) {
 
 	// Parse the configuration and determine if a broker and/or server exist.
 	config := parseConfig(*configPath, *hostname)
-	hasBroker := fileExists(config.Broker.Dir)
-	hasServer := fileExists(config.Data.Dir)
-	initializing := !hasBroker && !hasServer
+	initializing := !fileExists(config.Broker.Dir) && !fileExists(config.Data.Dir)
 
-	// Open broker if it exists or if we're initializing for the first time.
-	var b *messaging.Broker
-	var h *Handler
-	if hasBroker || (initializing && (*role == "combined" || *role == "broker")) {
-		b = openBroker(config.Broker.Dir, config.BrokerConnectionString())
-
-		// If this is the first time running then initialize a broker.
-		// Update the seed server so the server can connect locally.
-		if initializing {
-			if err := b.Initialize(); err != nil {
-				log.Fatalf("initialize: %s", err)
-			}
-		}
+	// Open broker, initialize or join as necessary.
+	b := openBroker(config.Broker.Dir, config.BrokerURL(), initializing, joinURLs)
 
-		// Start the broker handler.
+	// Start the broker handler.
+	var h *Handler
+	if b != nil {
 		h = &Handler{brokerHandler: messaging.NewHandler(b)}
-		go func() { log.Fatal(http.ListenAndServe(config.BrokerListenAddr(), h)) }()
-		log.Printf("Broker running on %s", config.BrokerListenAddr())
+		go func() { log.Fatal(http.ListenAndServe(config.BrokerAddr(), h)) }()
+		log.Printf("broker running on %s", config.BrokerAddr())
```

Comment:
Very minor: I wrote this log message I think. Now that I see it, I think "listening" is a better verb.